src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java

```diff
@@ -33,22 +31,29 @@
 import org.springframework.data.mapping.context.MappingContext;
 import org.springframework.lang.Nullable;
 
-public class CouchbaseTemplate implements CouchbaseOperations, ApplicationContextAware {
+/*
+ * @author Michael Nitschinger
+ * @author Michael Reiche
+ */
+
+public class CouchbaseTemplate implements CouchbaseOperations, ApplicationContextAware /*, ApplicationEventPublisherAware */ {
 
 	private final CouchbaseClientFactory clientFactory;
 	private final CouchbaseConverter converter;
 	private final PersistenceExceptionTranslator exceptionTranslator;
 	private final CouchbaseTemplateSupport templateSupport;
 	private final MappingContext<? extends CouchbasePersistentEntity<?>, CouchbasePersistentProperty> mappingContext;
 	private final ReactiveCouchbaseTemplate reactiveCouchbaseTemplate;
-	private @Nullable CouchbasePersistentEntityIndexCreator indexCreator;
-
+	private @Nullable
+	CouchbasePersistentEntityIndexCreator indexCreator;
+	//private ApplicationEventPublisher eventPublisher;// TODO: this field not needed?
 
 	public CouchbaseTemplate(final CouchbaseClientFactory clientFactory, final CouchbaseConverter converter) {
 		this.clientFactory = clientFactory;
 		this.converter = converter;
 		this.exceptionTranslator = clientFactory.getExceptionTranslator();
 		this.templateSupport = new CouchbaseTemplateSupport(converter);
+		// TODO: why doesn't the ReactiveCouchbaseTemplate use the same templateSupport ?
```

Comment:
still need this?

Reply:
It's not needed, but I still have that question. Why does doesn't  ReactiveCouchbaseTemplate use the same templateSupport as CouchbaseTemplate?

Reply:
removed